### PR TITLE
Lint fix for other modules

### PR DESCRIPTION
**Pull request by YifeiZuo01**
## Status
**READY**

## Migrations
NO

## Description
This is lint fix for modules:
- [x] init.py
- [x] common.py
- [x] erase.py
- [x] flash.py
- [x] main.py
- [x] reset.py
- [x] setup.py
- [x] tests:
* [x] test_erase.py
* [x] test_flash.py
* [x] test_main.py
* [x] test_reset.py
* [x] test_verify_flash.py

### CODE REVIEW

File: mbed_flasher/erase.py [hunk 1]
```diff
@@ -98,27 +115,33 @@ def runner(self, drive):
...
-                self.logger.debug("erase check timed out for %s" % drive[0])
+                self.logger.debug("erase check timed out for %s", drive[0])
                 break
-    
+
     def erase_board(self, mount_point, serial_port, no_reset):
+        """
+        :param mount_point: mount point
+        :param serial_port: serial port
+        :param no_reset: earse with/without reset
```
Comment (jonikula) [inline mbed_flasher/erase.py]: Typo here.

File: mbed_flasher/erase.py [hunk 2]
```diff
@@ -98,27 +115,33 @@ def runner(self, drive):
...
             self.logger.warn("Experimental feature, might not do anything!")
             self.logger.info("erasing device")
-            with open(join(mount_point, 'ERASE.ACT'), 'wb') as new_file:
+            with open(join(mount_point, 'ERASE.ACT'), 'wb'):
                 pass
-            t = Thread(target=self.runner, args=([mount_point, 'ERASE.ACT'],))
-            t.start()
-            while t.is_alive():
-                t.join(0.5)
+            thread = Thread(target=self.runner, args=([mount_point, 'ERASE.ACT'],))
```
Comment (jonikula) [inline mbed_flasher/erase.py]: thread is not a good variable name (shadows thread module name)

File: mbed_flasher/erase.py (hunk 1 above)
Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: what typo?

Comment (jonikula) [inline mbed_flasher/erase.py]: earse -> erase

File: mbed_flasher/erase.py [hunk 3]
```diff
@@ -42,29 +44,41 @@ class Erase(object):
...
-            devices = Flasher.get_available_devices()
+        for flasher in self.flashers:
+            devices = flasher.get_available_devices()
             available_devices.extend(devices)
         return available_devices
 
     @staticmethod
     def __get_flashers():
+        """
+        :return: list of avaliable flashers
```
Comment (juhhov) [inline mbed_flasher/erase.py]: *available

File: mbed_flasher/flash.py [hunk 4]
```diff
@@ -29,79 +29,119 @@
...
-        for Flasher in AvailableFlashers:
-            if Flasher.name.lower() == flasher.lower():
-                return Flasher
-        else:
-            return EXIT_CODE_REQUESTED_FLASHER_DOES_NOT_EXIST
+        """
+        :param flasher: None, if not given a flasher
+        :return: return available flasher if found, otherwise return exit code
+        """
+        from mbed_flasher.flashers import AvailableFlashers
```
Comment (juhhov) [inline mbed_flasher/flash.py]: Is there really some benefit to do this import here instead of the usual place?

File: mbed_flasher/flash.py [hunk 5]
```diff
@@ -29,79 +29,119 @@
...
+        :param platform_name: platform name
+        :return:
+        """
+        if platform_name not in self.supported_targets:
             raise NotImplementedError("Flashing %s is not supported" % platform_name)
 
-        for Flasher in self.FLASHERS:
-            if platform_name in self.SUPPORTED_TARGETS:
+        # pylint: disable=invalid-name
+        for Flasher in self._flashers:
```
Comment (juhhov) [inline mbed_flasher/flash.py]: Why is this pylint disable?

File: mbed_flasher/flash.py (hunk 4 above)
Comment (YifeiZuo01) [inline mbed_flasher/flash.py]: not really, the logic here is going to be changed soon, maybe I should add todo.

File: mbed_flasher/main.py [hunk 6]
```diff
@@ -106,26 +112,29 @@ def __init__(self, args=None):
...
+            for name in files:
+                if str(name).find('_mbed-flasher.txt') != -1:
+                    if old_logs > time.mktime(time.strptime(str(name).split('_')[0],
+                                                            "%Y%m%d-%H%M%S")):
                         files_to_be_removed.append(str(os.path.join(root, name)))
-                    else:
-                        pass
+                elif str(name).find('mbed-flasher.log') != -1:
+                    files_to_be_removed.append(str(os.path.join(root, name)))
+                else:
```
Comment (juhhov) [inline mbed_flasher/main.py]: Redundant else?

File: mbed_flasher/main.py [hunk 7]
```diff
@@ -222,61 +299,73 @@ def subcmd_flash_handler(self, args, logger):
...
+        if 'all' in args.tid:
+            retcode = flasher.flash(build=args.input, target_id='all',
+                                    platform_name=args.platform_name,
+                                    method=args.method, no_reset=args.no_reset)
+
+        if len(available) <= 0:
+            print("Could not find any connected device")
+            return EXIT_CODE_DEVICES_MISSING
+
+        if isinstance(args.tid, list):
```
Comment (juhhov) [inline mbed_flasher/main.py]: This if-else looks like it could use some refactoring.

File: test/test_main.py [hunk 8]
```diff
@@ -17,13 +17,12 @@
```
Comment (juhhov) [inline test/test_main.py]: Could add ```# pylint:disable=missing-docstring``` here to not have to add those function docstrings.

File: mbed_flasher/main.py (hunk 7 above)
Comment (YifeiZuo01) [inline mbed_flasher/main.py]: it looks so. I did something to avoid too many if-else, these parts is remaining to be as it was.  I will keep work on it.

Comment (YifeiZuo01) [inline mbed_flasher/main.py]: I refactored this part in [this commit](https://github.com/ARMmbed/mbed-flasher/pull/55/commits/c68b0d92e0f107d1d30383fdbd25f812d646ad05)
could you please check if it's correct and what else can be improved.

Comment (juhhov) [inline mbed_flasher/main.py]: Maybe the below part could be separated into its own function:
```
if device['target_id'] == item or device['target_id'].startswith(item):
    if device['target_id'] not in target_ids_to_flash:
        target_ids_to_flash.append(device['target_id'])

    if 'platform_name' in device and device['platform_name'] not in available_platforms:
        available_platforms.append(device['platform_name'])
```

That part is there twice with minimal difference.

File: mbed_flasher/flash.py [hunk 9]
```diff
@@ -29,79 +30,115 @@
...
+        :return: target
+        """
+        for target in target_list:
             if platform_name == target['platform_name']:
                 return target
         raise KeyError("platform_name: %s not found" % platform_name)
 
-    def flash_multiple(self, build, platform_name, method='simple', target_ids_or_prefix='', no_reset=None):
+    # Todo code improvement
+    # pylint: disable=too-many-nested-blocks, too-many-branches
```
Comment (YifeiZuo01) [inline mbed_flasher/flash.py]: @juhhov  first one here, do you think if we need improvement?

File: mbed_flasher/reset.py [hunk 10]
```diff
@@ -74,14 +89,16 @@ def reset_board(self, serial_port):
                 self.logger.error("reset failed")
                 return EXIT_CODE_SERIAL_RESET_FAILED
         port.close()
-        
+
+    # Todo code improvement
+    # pylint: disable=too-many-nested-blocks, too-many-branches
     def reset(self, target_id=None, method=None):
```
Comment (YifeiZuo01) [inline mbed_flasher/reset.py]: @juhhov  second one

File: mbed_flasher/erase.py [hunk 11]
```diff
@@ -130,13 +153,16 @@ def erase_board(self, mount_point, serial_port, no_reset):
             print("Selected device does not support erasing through DAPLINK")
             return EXIT_CODE_IMPLEMENTATION_MISSING
 
+    # Todo code improvement
+    # pylint: disable=too-many-branches, too-many-nested-blocks
     def erase(self, target_id=None, no_reset=None, method=None):
```
Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: @juhhov third one

File: mbed_flasher/flash.py (hunk 9 above)
Comment (juhhov) [inline mbed_flasher/flash.py]: Yes please.

File: mbed_flasher/reset.py (hunk 10 above)
Comment (juhhov) [inline mbed_flasher/reset.py]: This one should be refactored also.

File: mbed_flasher/erase.py (hunk 11 above)
Comment (juhhov) [inline mbed_flasher/erase.py]: This one too.

File: mbed_flasher/reset.py (hunk 10 above)
Comment (YifeiZuo01) [inline mbed_flasher/reset.py]: improvements added.

File: mbed_flasher/erase.py (hunk 11 above)
Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: changes added.

File: mbed_flasher/flash.py [hunk 12]
```diff
@@ -140,101 +154,57 @@ def flash_multiple(self, build, platform_name,
...
         self.logger.debug(device_mapping_table)
 
-        # python 3 compatibility
-        # pylint: disable=superfluous-parens
-        print('Going to flash following devices:')
+        self.logger.info('Going to flash following devices:')
         for item in device_mapping_table:
-            print(item['target_id'])
-        retcodes = 0
-        if method == 'pyocd' and platform.system() != 'Windows':
```
Comment (YifeiZuo01) [inline mbed_flasher/flash.py]: this removed, where the refactored code does this actions?

Comment (juhhov) [inline mbed_flasher/flash.py]: That was obsolete code.

If you look at the original code you'll notice that both branches of that if-else had basically same functionality.

Comment (YifeiZuo01) [inline mbed_flasher/flash.py]: yes, it is

File: mbed_flasher/erase.py [hunk 13]
```diff
@@ -131,66 +154,77 @@ def erase_board(self, mount_point, serial_port, no_reset):
...
+            return EXIT_CODE_TARGET_ID_MISSING
+
+        targets_to_erase = self.prepare_target_to_erase(target_id, available_devices)
+
+        if len(targets_to_erase) <= 0:
+            print("Could not map given target_id(s) to available devices")
+            return EXIT_CODE_COULD_NOT_MAP_TO_DEVICE
+
+        for item in targets_to_erase:
+            if item['platform_name'] != 'K64F':
```
Comment (jonikula) [inline mbed_flasher/erase.py]: Is this restriction ok? Is K64F the only board that is supported?

File: mbed_flasher/flash.py [hunk 14]
```diff
@@ -29,163 +29,184 @@
...
+            ret = self.flash(build=build,
+                             target_id=device['target_id'],
+                             platform_name=None,
+                             device_mapping_table=device_mapping_table,
+                             method=method,
+                             no_reset=no_reset)
+            if ret == 0:
+                self.logger.debug("dev#%i -> SUCCESS", i)
+            else:
+                self.logger.warning("dev#%i -> FAIL :(", i)
```
Comment (jonikula) [inline mbed_flasher/flash.py]: Should this emote be removed from these prints?

File: mbed_flasher/erase.py (hunk 13 above)
Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: the changes follows the code as it was. @VeliMattiLahtela any idea?

Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: for the current situation, it is the only one supported, maybe some more in the future.

File: mbed_flasher/flash.py (hunk 14 above)
Comment (YifeiZuo01) [inline mbed_flasher/flash.py]: fixed

**Pull request merged**